test/unit/storage/DataAccessorBasedStore.test.ts

```diff
@@ -216,6 +218,12 @@ describe('A DataAccessorBasedStore', (): void => {
       );
       expect(result.metadata.contentType).toBe(INTERNAL_QUADS);
     });
+
+    it('throws a 304 if the request is a read type error.', async(): Promise<void> => {
+      const resourceID = { path: root };
+      const conditions = new BasicConditions({ notMatchesETag: [ '*' ]});
+      await expect(store.getRepresentation(resourceID, undefined, conditions)).rejects.toThrow(NotModifiedHttpError);
+    });
```

Comment:
Can we also have a test where there are conditions present but no error is thrown? Just to be complete.

Reply:
Think I covered all this - was just confused about what the new test should resolve to be.. I put resolves.toBeTruthy but if you know what it should be then I'll just merge that and it should be good to go.

Reply:
To be completely correct you should check that the output matches the data that we are using to mock what is stored at the root. But in practice it doesn't matter that much so this is fine.